worlds/mlss/Rom.py

```diff
@@ -0,0 +1,587 @@
+        patch.write_token(
+            APTokenTypes.WRITE,
+            0x3AC728,
+            bytes([0x89, 0x13, 0x0, 0x10, 0xF, 0x08, 0x1])
+        )
+        patch.write_token(
+            APTokenTypes.WRITE,
+            0x3AC56C,
+            bytes([0x49, 0x16, 0x0, 0x8, 0x8, 0x08, 0x1])
+        )
+
+    if world.options.scale_stats:
+        patch.write_token(
+            APTokenTypes.WRITE,
+            0xD00002,
+            bytes([0x1])
+        )
+
+    if world.options.xp_multiplier:
+        patch.write_token(
+            APTokenTypes.WRITE,
+            0xD00003,
+            bytes([world.options.xp_multiplier.value])
+        )
+
+    if world.options.tattle_hp:
+        patch.write_token(
+            APTokenTypes.WRITE,
+            0xD00000,
+            bytes([0x1])
+        )
+
+    if world.options.music_options == 2:
+        patch.write_token(
+            APTokenTypes.WRITE,
+            0x19B118,
+            bytes([0x0, 0x25])
+        )
+
+    if world.options.randomize_backgrounds:
+        all_enemies = Enemies.enemies + Enemies.bosses
+        for address in all_enemies:
+            patch.write_token(
+                APTokenTypes.WRITE,
+                address + 3,
+                bytes([world.multiworld.per_slot_randoms[world.player].randint(0x0, 0x26)])
+            )
+
+    for location_name in location_table.keys():
+        if (world.options.skip_minecart and "Minecart" in location_name and "After" not in location_name) or (
+                world.options.castle_skip and "Bowser" in location_name) or (
+                world.options.disable_surf and "Surf Minigame" in location_name) or (
+                world.options.harhalls_pants and "Harhall's" in location_name):
+            continue
+        if (world.options.chuckle_beans == 0 and "Digspot" in location_name) or (
+                world.options.chuckle_beans == 1 and location_table[location_name] in hidden):
+            continue
+        if not world.options.coins and "Coin" in location_name:
+            continue
+        location = world.multiworld.get_location(location_name, world.player)
+        item = location.item
+        address = [address for address in all_locations if address.name == location.name]
+        item_inject(world, patch, location.address, address[0].itemType, item)
+        if "Shop" in location_name and "Coffee" not in location_name and item.player != world.player:
+            desc_inject(world, patch, location, item)
+
+    swap_colors(world, patch, colors[world.options.mario_color], 0)
+    swap_colors(world, patch, colors[world.options.luigi_color], 1)
+    swap_pants(world, patch, cpants[world.options.mario_pants], 0)
+    swap_pants(world, patch, cpants[world.options.luigi_pants], 1)
+
+    patch.write_file("token_data.bin", patch.get_token_binary())
+
+
+def swap_colors(world: "MLSSWorld", patch: MLSSProcedurePatch, color: str, bro: int):
+    temp = pkgutil.get_data(__name__, "colors/" + color + ".txt")
```

Comment:
core devs would want me to say here that .py files would be much faster than .txt files, if you can make it work

At least it's for an optional cosmetic option so it's probably not so bad. But yeah there is a movement right now against loading data files via file io because file io cannot be optimised when this is compiled to an .exe

*Low priority according to me, probably high priority according to some people*